### PR TITLE
Free OS memory in QueryNode and IndexNode

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

issues: #12218

### CODE REVIEW

File: internal/querynode/task.go [hunk 1]
```diff
@@ -793,6 +795,9 @@ func (r *releaseCollectionTask) Execute(ctx context.Context) error {
 	}
 	r.node.historical.removeGlobalSegmentIDsByCollectionID(r.req.CollectionID)
 
+	runtime.GC()
```
Comment (xiaofan-luan) [inline internal/querynode/task.go:798]: debug.FreeOSMemory seems to trigger runtime.GC automatically

Comment (bigsheeper) [reply internal/querynode/task.go:798]:
```golang
//go:linkname runtime_debug_freeOSMemory runtime/debug.freeOSMemory
func runtime_debug_freeOSMemory() {
	GC()
	systemstack(func() { mheap_.scavengeAll() })
}
```

you're right

Comment (bigsheeper) [reply internal/querynode/task.go]: removed

**Pull request merged**